### PR TITLE
Update 3rd parties, globjects induce changes to Viewer.

### PR DESCRIPTION
**Pull request by dlyr**
Use WindowQt as base viewer widget, copied and edited from glbinding examples so next updates will be easy.

* **Please check if the PR fulfills these requirements**
- [X] Is the Pull-Request done against the right branch:
  - `master`: for hotfixes not impacting the API
  - `master-v1`: for changes related to the Radium Stable Release V1.
- [X] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Update 3rd parties

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**
Add a base class for the viewer as close as possible to the one found in globject example, to ease futur update.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:
This PR has to be tested on mac/windows since it changes context and gl initialization. As the base class from globject should work on these plateform, I hope nothing is broken.

### CODE REVIEW

File: src/GuiBase/Viewer/Viewer.hpp [hunk 1]
```diff
@@ -69,24 +71,12 @@ class RA_GUIBASE_API Viewer : public QWindow {
     /// create gizmos
     void createGizmoManager();
 
-    /**
-     * Make the OpenGL context associated with the viewer the current context.
-     */
-    void makeCurrent();
```
Comment (nmellado) [inline src/GuiBase/Viewer/Viewer.hpp]: Aren't these methods potentially useful for third party component ? 
For instance, to active the GL context before doing some GL operations ?

File: src/GuiBase/Viewer/WindowQt.hpp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+#include <atomic>
+
+class QSurfaceFormat;
+class QOpenGLContext;
+
+namespace Ra {
+namespace Gui {
+
+class RA_GUIBASE_API WindowQt : public QWindow {
```
Comment (nmellado) [inline src/GuiBase/Viewer/WindowQt.hpp]: Could you mention this class comes from globjects examples ?
```suggestion
/*!
 * \brief Base class for OpenGL widgets, compatble with Qt and globjects/glbindings.
 * \note This class has been extracted from globject example _add-url-here_
 */
class RA_GUIBASE_API WindowQt : public QWindow {
```

File: src/GuiBase/Viewer/Viewer.cpp [hunk 3]
```diff
@@ -247,21 +254,15 @@ void Gui::Viewer::intializeRenderer( Engine::Renderer* renderer ) {
...
 
-void Gui::Viewer::resizeGL( int width_, int height_ ) {
-    if ( isExposed() )
-    {
-        // Renderer should have been locked by previous events.
-        m_context->makeCurrent( this );
-
-        // see issue #261 Qt Event order and default viewport management (Viewer.cpp)
-        // https://github.com/STORM-IRIT/Radium-Engine/issues/261
-#ifndef OS_MACOS
```
Comment (nmellado) [inline src/GuiBase/Viewer/Viewer.cpp]: Are these changes validated on MacOS ?

File: src/GuiBase/Viewer/Viewer.hpp (hunk 1 above)
Comment (dlyr) [reply src/GuiBase/Viewer/Viewer.hpp]: they are in WindowQt, so available to third parties

File: src/GuiBase/Viewer/Viewer.cpp (hunk 3 above)
Comment (MathiasPaulin) [reply src/GuiBase/Viewer/Viewer.cpp]: Not able to test, there is a dylib problem with assimp. I'll try to fix it.

Comment (dlyr) [reply src/GuiBase/Viewer/Viewer.cpp]: Using DYLD_LIBRARY_PATH, added to README.md

**Pull request merged**